xgboost_ray/main.py

```diff
@@ -342,27 +345,35 @@ def _train(params: Dict,
     # Train
     fut = [
         actor.train.remote(rabit_args, params, dtrain, evals, *args, **kwargs)
-        for (actor, _) in actors
+        for actor in actors
     ]
 
     callback_returns = [list() for _ in range(len(actors))]
     try:
-        ready, not_ready = ray.wait(fut, num_returns=len(fut), timeout=0)
+        not_ready = fut
         while not_ready:
-            for i, (actor, queue) in enumerate(actors):
+            if queue:
                 while not queue.empty():
                     item = queue.get()
                     if isinstance(item, Callable):
                         item()
                     else:
                         callback_returns[i].append(item)
-            ready, not_ready = ray.wait(fut, num_returns=len(fut), timeout=0)
+            ready, not_ready = ray.wait(not_ready, timeout=0)
```

Comment:
This is a key difference - this makes it so that we do not wait for all workers to finish - rather, every worker that finishes, we call `ray.get` on its result. This will allow error messages to propagate properly, even before the rest of the workers are done.

Reply:
That makes sense, as other workers are waiting for the killed worker to finish before they finish themselves. That explains why it stalled